### ISSUE TITLE
HTTP API

### ISSUE THREAD

**Issue by martinthomson** [w3c/attribution#31]
In deploying ARA, Google discovered that there are a great many intermediaries who are incapable of running JavaScript and many others that prefer not to.  There's a lot of things like tracking pixels, with the HTTP interactions to load those pixels generating measurement signals as a side effect.

Adding HTTP header fields to those requests that trigger the equivalent of the APIs might make the entire system easier for these systems to consume.

There are some serious difficulties involved with this mode of interaction.  Responses will contain instructions to the browser to save impressions or measure conversions.  New requests will be needed in order to deliver the results of those interactions.  That was true for ARA, but the delays meant that new requests were inevitable.  Here, the new interactions would need to be synthesized somehow.  Using a redirect is one possible option.

**Comment by apasel422** [comment]
@martinthomson If you think it's worthwhile, I'm happy to draft a PR to at least define the structured-header representation of a hypothetical `Measure-Conversion` header.

**Comment by martinthomson** [comment]
I don't think that that is the challenge here.  The difficulty is in how the browser responds to the header and gets the result back to the site that wanted it.  There are some sketches, but those stretch the bounds of what the protocol is suited for.  From my perspective, getting that settled is important.  Just to give a sense of the challenge here, we'd need to work out what the CORS disposition of any such fetch would be.  (You might have a better sense of what works there, but my understanding is that it isn't trivial to pilot a course through the shark-infested waters that surround Fetch.)

**Comment by bmayd** [comment]
@martinthomson re: 
> I'd rather spend that time on advancing an MVP.

I agree and am supportive of focusing on an MVP. That said, if someone has bandwidth and the inclination to consider a non-script approach there's a high probability it would be time well spent.
